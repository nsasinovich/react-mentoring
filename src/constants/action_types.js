export default {
    CHANGE_SORT: 'CHANGE_SORT',
    UPDATE_SEARCH_INPUT: 'UPDATE_SEARCH_INPUT',
};
