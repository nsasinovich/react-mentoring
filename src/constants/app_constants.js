export const PosterSizes = {
    LARGE: 'poster-large',
    MEDIUM: 'poster-medium',
};

export const FilterOptions = {
    GENRE: 'genres',
    TITLE: 'title',
};

export const SortingOptions = {
    RELEASE: {
        name: 'release date',
        sortField: 'year',
    },
    RATING: {
        name: 'rating',
        sortField: 'vote_count',
    },
};

export const INITIAL_STATE = {
    results: [],
    resultsCount: 0,
    searchInput: '',
    selectedSort: SortingOptions.RATING,
    selectedFilter: FilterOptions.TITLE,
    selectedMovieDetails: null,
};
