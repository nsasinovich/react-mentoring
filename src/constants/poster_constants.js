export const PosterSizes = {
    LARGE: 'poster-large',
    MEDIUM: 'poster-medium',
};
