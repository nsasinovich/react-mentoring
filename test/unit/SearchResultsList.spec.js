import { shallow } from 'enzyme';
import React from 'react';
import SearchResultsList from 'src/components/search_results_list/search_results_list';
import NoResults from 'src/components/no_results/no_results';
import AssetTile from 'src/components/asset_tile/asset_tile';
import { SortingOptions } from 'src/constants/app_constants';

const { sortingField } = SortingOptions.RATING;

describe('<SearchResultsList />', () => {
    it('should render SearchResultsList and match snapshot', () => {
        const wrapper = shallow(<SearchResultsList sortField={sortingField} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render NoResults if results array is empty', () => {
        const wrapper = shallow(<SearchResultsList results={[]} sortField={sortingField} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.contains(<NoResults />)).toBe(true);
    });

    it('should render AssetTiles if results are passed', () => {
        const resultAssets = [
            { title: 'Asset 1', year: 1995, rating: 1 },
            { title: 'Asset 2', year: 1994, rating: 3.53 },
            { title: 'Asset 3', year: 1993, rating: 3.55 },
            { title: 'Asset 4', year: 1989, rating: 4 },
            { title: 'Asset 5', year: 1988, rating: 5 },
        ];

        const wrapper = shallow(<SearchResultsList results={resultAssets} sortField={sortingField} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(AssetTile)).toHaveLength(resultAssets.length);
    });
});
