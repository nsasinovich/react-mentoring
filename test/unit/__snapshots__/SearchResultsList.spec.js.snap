// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultsList /> should render AssetTiles if results are passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResultsList
    dispatch={[Function]}
    results={Array []}
    sortField="vote_count"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="search-results-list"
      >
        <NoResults />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <NoResults />,
        "className": "search-results-list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="search-results-list"
        >
          <NoResults />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NoResults />,
          "className": "search-results-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<SearchResultsList /> should render NoResults if results array is empty 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResultsList
    dispatch={[Function]}
    results={Array []}
    sortField="vote_count"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="search-results-list"
      >
        <NoResults />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <NoResults />,
        "className": "search-results-list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="search-results-list"
        >
          <NoResults />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NoResults />,
          "className": "search-results-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<SearchResultsList /> should render SearchResultsList and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchResultsList
    dispatch={[Function]}
    results={Array []}
    sortField="vote_count"
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="search-results-list"
      >
        <NoResults />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <NoResults />,
        "className": "search-results-list",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="search-results-list"
        >
          <NoResults />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NoResults />,
          "className": "search-results-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
